From ac0fca8559c2384240a00599a46816bbb5afb93f Mon Sep 17 00:00:00 2001
From: Thomas Dickerson <elfprince13@gmail.com>
Date: Tue, 11 Mar 2025 14:07:34 -0400
Subject: [PATCH] Allow empty configuration values to be supplied from
 environment

For example it may be desirable for `CC_OPT_FLAGS` to be empty.
---
 configure.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.py b/configure.py
index ec04fcfdd0cc67..ac19b856d7c914 100644
--- a/configure.py
+++ b/configure.py
@@ -529,7 +529,9 @@ def get_from_env_or_user_or_default(environ_cp, var_name, ask_for_var,
     string value for var_name
   """
   var = environ_cp.get(var_name)
-  if not var:
+  # an intentionally empty value in the
+  # environment is not the same as no value
+  if var is None:
     var = get_input(ask_for_var)
     print('\n')
   if not var:
